ingest/anndata_.py

```diff
@@ -35,3 +44,66 @@ def validate(self):
         except ValueError:
             return False
 
+    @staticmethod
+    def generate_cluster_header(adata, clustering_name):
+        """
+        Based on clustering dimensions, write clustering NAME line to file
+        """
+        dim2 = ['NAME', 'X', 'Y']
+        dim3 = dim2 + ['Z']
+        clustering_dimension = adata.obsm[clustering_name].shape[1]
```

Comment:
I see this a couple places - should this maybe be fetched once (in a constructor maybe) and passed as a value?

Reply:
It's in two places (if it were more, I'd consider changing it) but since the dimension is data implicit in the adata object that's passed in, extracting that info and passing it in seems to open up a possibility for error rather than any particular advantage. Happy to discuss further why it'd be useful to extract and pass in. I can certainly make the change during the upcoming metadata work.